source/server/overload_manager_impl.cc

```diff
@@ -86,7 +86,7 @@ class ThresholdTriggerImpl final : public OverloadAction::Trigger {
     const OverloadActionState state = actionState();
     state_ =
         value >= threshold_ ? OverloadActionState::saturated() : OverloadActionState::inactive();
-    return state.value() != actionState().value();
+    return state.value().value() != actionState().value().value();
```

Comment:
If you are going to fall back to this, I don't think we've actually improved anything, right?

2 other options I've thought of:
 * provide operator== but make its implementation approximate, e.g. `abs(a - b) < K*(max-min)`
 * provide methods that invert a value between min and max without losing exactness

Reply:
This makes the floating point equality comparison visible instead of hiding it behind `UnitFloat::operator==`. I could see the argument for comparing with a fudge factor but my preference would be to do that here since it's a statement about how precise the OverloadManagerImpl wants to be, not about UnitFloat values.

Reply:
In this file, on line 63, you set the value to 1-value.

If this is not exact, then we have a potential problem, right?

Reply:
And if you still don't like the approximate comparison (which I can understand) it looks like it would be easy enough for you to replace the 1-value pattern with an exactness-preserving inversion operation.

I think if you just pull the floats out and compare them then this is a bit of a leaky abstraction, right?

Reply:
I see the 1-value on line 63, but I'm having trouble understanding the potential problem. Are you worried that `UnitFloat(1 - UnitFloat::max())` is not equal to `UnitFloat::min()` and vice-versa?

> this is a bit of a leaky abstraction, right?

UnitFloat only guarantees that if you call `.value()`, the value you get out is in the range [0, 1], and that it is as close as possible to the float provided in the constructor.

Reply:
Talked offline with @jmarantz; added comments and `UnitFloat::invert()` for the 1-pattern, and added back in `UnitFloat::operator==` and `!=`